tests/inspector/cli-codegen-1.spec.ts

```diff
@@ -539,7 +539,7 @@ test.describe('cli codegen', () => {
     expect(sources.get('Playwright Test').text).toContain(`
   // Click text=link
   await page.click('text=link');
-  expect(page.url()).toBe('about:blank#foo');`);
+  expect(page).toHaveURL('about:blank#foo');`);
```

Comment:
The previous assertion is more accurate as by the time click on the link finishes the page should already be on the new URL and there is no point in retrying that check (which is what toHaveURL would do).

Reply:
Interesting. In one of my projects where I am generating tests for a Next.js application, I have encountered this problem multiple times. Maybe there is some other bug, then?

Reply:
`click()` will autowait for the first navigation if it happens, if there is a subsequent one the url may change again before the place where you check it, in which case toHaveURL would help but I'm not sure this should be default behavior.

Reply:
In my case there has always been only one navigation after the click.

Here's a proof of concept of the problem: https://github.com/nygrenh/playwright-wait-for-url-problem-poc

It has a page with a link that navigates to another one (https://github.com/nygrenh/playwright-wait-for-url-problem-poc/blob/0b1bce60db4861a85eef53ff35dc438ac8273e24/pages/index.js#L10) and a test (https://github.com/nygrenh/playwright-wait-for-url-problem-poc/blob/0b1bce60db4861a85eef53ff35dc438ac8273e24/tests/example.spec.js). Instructions for running the POC in README.

Reply:
In your case the navigation is asynchronous and the generated code would actually be:
```js
  await Promise.all([
    page.waitForNavigation(/*{ url: 'http://localhost:3000/page-2' }*/),
    page.click('text=Page 2')
  ]);
```
and this is what you should have in your test to make it pass. So this codegen change would not affect it at all.